### PR TITLE
TMEDIA-653 ci : Use uniform pattern for caching dependencies across Github Actions.

### PR DESCRIPTION
**Pull request by ecarlisle**
## Description
Ensure github action caching uniformity 

## Jira Ticket
- [TMEDIA-653](https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TMEDIA-653)

## Acceptance Criteria
Github actions caching is uniformly done throughout blocks, themes components, and engine theme sdk

Write a doc in the themes internal space around caching and best practices there 

Make sure that the cache is reset when there’s a change to the package-lock and used when there’s no change to the package-lock

## Test Steps
- See example in lerna playground where this structure and investigation works https://github.com/WPMedia/themes-lerna-playground/pull/15

### Before
- Unnecessary caching and potentially not enough installing

### After
- Unnecessary caching and potentially not enough installing

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps above are working
- [ ] Confirmed there are no linter errors
- [ ] Confirmed this PR has reasonable code coverage
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

### CODE REVIEW

File: .github/workflows/beta-publish.yml [hunk 1]
```diff
@@ -14,28 +14,19 @@ jobs:
...
-          key: node-modules-${{ hashFiles('package-lock.json') }}
-
       - name: Setup Node
         uses: actions/setup-node@v1
         with:
           node-version: "12"
           registry-url: "https://npm.pkg.github.com"
 
-      - name: Clean install dependencies if lockfile changed
+      - name: Cache Node files
```
Comment (JackHowa) [inline .github/workflows/beta-publish.yml:23]: needs an `id: cache`

**Review by JackHowa — CHANGES_REQUESTED** [review]
Need to use id: cache. is this ready for review?

File: .github/workflows/beta-publish.yml (hunk 1 above)
Comment (JackHowa) [reply .github/workflows/beta-publish.yml:23]: otherwise the npm ci will always run

Comment (JackHowa) [reply .github/workflows/beta-publish.yml:23]: https://github.com/WPMedia/themes-lerna-playground/pull/15/files#diff-3928becfcdf7981d7c15d4a8714a11384ecbdbda50fa5abb08cd04e456678414R24

Comment (ecarlisle) [reply .github/workflows/beta-publish.yml:23]: Thanks! got it!

**Review by JackHowa — DISMISSED** [review]

**Comment by matthewroach** [comment]
Both actions that run on this pull request that require caching are both failing due to issues with restored caches. Is there further work needed to ensure the cache is working consistently?

**Pull request merged**